pennylane/compiler/qjit_api.py

```diff
@@ -568,3 +603,28 @@ def __call__(self, *init_state):
             args = fn_res if len(args) > 1 else (fn_res,) if len(args) == 1 else ()
 
         return fn_res
+
+    def _call_capture_enabled(self, *init_state):
+
+        import jax  # pylint: disable=import-outside-toplevel
+
+        for_loop_prim = _get_for_loop_qfunc_prim()
+
+        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)(0, *init_state)
```

Comment:
What is the first argument `0` for `jax.make_jaxpr(self.body_fn)` here for? 😅

Reply:
This is because of how the syntax of `qml.for` works. This `0` corresponds to the index of the for loop (it does not matter the exact value as long as it is an integer since the value is established at runtime). By convention, it always corresponds to the first argument of the loop body function, and we must capture it here along with the other arguments.

Reply:
`0` forces the creation of the jaxpr with the first argument being a single integer. We could equivalently use any other integer, or `self.lower_bound`, but using `0` is a simple enough way to construct the correct jaxpr.

Reply:
Ahh okay, I was just confused about the choice for using `0` and not `self.lower_bound`. Thanks for clarifying!